crypto/crypto.go

```diff
@@ -34,11 +35,11 @@ func Sign(key SigningKey, message []byte) []byte {
 	return ed25519.Sign(ed25519.PrivateKey(key), message)
 }
 
-func Verify(key SigningKey, message, sig []byte) bool {
-	pk, ok := ed25519.PrivateKey(key).Public().(ed25519.PublicKey)
-	if !ok {
-		return false
-	}
+func SigningKeyToPublic(key SigningKey) []byte {
+	return ed25519.PrivateKey(key).Public().(ed25519.PublicKey)
+}
+
+func Verify(pk, message, sig []byte) bool {
 	return ed25519.Verify(pk, message, sig)
 }
```

Comment:
I think your suggestion is compatible with the above change,

We'd just make these two additional changes,

```
-func (key SigningKey) Public() []byte {
+func (key SigningKey) Public() VerifKey {
```

```
-func Verify(pk, message, sig []byte) bool {
-func (pk VerifKey) Verify(message, sig []byte) bool {
```

where `Verify` is now a method of the `VerifKey` type.